### ISSUE TITLE
Parsing error: Unexpected token )

### ISSUE THREAD

**Issue by shenhao98** [vuejs/eslint-plugin-vue#2468]
**Checklist**

- [y ] I have tried restarting my IDE and the issue persists.
- [y ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 9.3.0** 
- **eslint-plugin-vue version: 9.26.0** 
- **Vue version: 3.4.21** 
- **Node version: v18.20.0** 
- **Operating System: win10 64** 

**Please show your full configuration:**

```json5
// eslint.config.js
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginVue from "eslint-plugin-vue";

export default [
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  ...pluginVue.configs["flat/strongly-recommended"],
  {
    languageOptions: { globals: globals.browser },
    rules: {
      "@typescript-eslint/no-unused-vars": "warn",
      "no-unused-vars": "warn",
      "no-undef": "error"
    },
  },
];
```

**What did you do?**	

```js

```

**What did you expect to happen?  
 **

**What actually happened?**

![image](https://github.com/vuejs/eslint-plugin-vue/assets/55233979/02c0c0bc-e34f-4aff-9ca3-139784d8a0ef)
**Repository to reproduce this issue**

**Comment by FloEdelmann** [comment]
You need to tell eslint-parser-vue that you are using TypeScript: https://eslint.vuejs.org/user-guide/#what-is-the-use-the-latest-vue-eslint-parser-error

**Comment by mrchar** [comment]
@shenhao98 I found the ansower at [typescript-eslint FAQ](https://typescript-eslint.io/troubleshooting/#i-am-running-into-errors-when-parsing-typescript-in-my-vue-files)
